mesonbuild/coredata.py

```diff
@@ -140,20 +141,20 @@ def set_value(self, newvalue: T.Any) -> bool:
         return self.value != oldvalue
 
 class UserStringOption(UserOption[str]):
-    def __init__(self, description: str, value: T.Any, yielding: bool = DEFAULT_YIELDING,
+    def __init__(self, name: str, description: str, value: T.Any, yielding: bool = DEFAULT_YIELDING,
                  deprecated: T.Union[bool, str, T.Dict[str, str], T.List[str]] = False):
-        super().__init__(description, None, yielding, deprecated)
+        super().__init__(name, description, None, yielding, deprecated)
         self.set_value(value)
 
     def validate_value(self, value: T.Any) -> str:
         if not isinstance(value, str):
-            raise MesonException('Value "%s" for string option is not a string.' % str(value))
+            raise MesonException(f'Option "{self.name}" value "{value}" is not a string.')
```

Comment:
This reads a bit stiff to me, I think either adding a `:` after the option name (`Option "{self.name}: ...`), or something like: `The value of option "{self.name}" is "{value}", which is not a string` would read a bit easier.